localstack/services/plugins.py

```diff
@@ -12,11 +16,91 @@
 # set up logger
 LOG = logging.getLogger(__name__)
 
-# map of service plugins, mapping from service name to plugin details
-SERVICE_PLUGINS = {}
+# TODO: Define interfaces for ServiceLifecycle, SchemaValidation, SecurityEnforcement, etc.
+#       to achieve composable ServicePlugins.
+#  Ultimately, the plugin metamodel will allow to easily add new ServicePlugins that consist of:
+#     - optional initializer (e.g., download dependencies, apply patches)
+#     - service lifecycle (start, stop, pause)
+#     - health check
+#     - state manager for persistent state (potentially composite)
+#     - request/schema validator (future work)
+#     - security interceptor (future work)
+#     ...
 
 # maps service names to health status
-STATUSES = {}
+STATUSES: Dict[str, Dict] = {}
+
+
+# ---------------------------
+# STATE SERIALIZER INTERFACE
+# ---------------------------
+
+
+class PersistenceContext:
+    def __init__(self, state_dir: str = None, lock: rwlock.RWLockable = None):
+        # state dir (within DATA_DIR) of currently processed API in local file system
+        self.state_dir: str = state_dir
```

Comment:
in general it makes more sense to me to add type hints to member variables in the class declaration. that way you can get the type hints through the `__annotations__` magic var (which can be helpful for docs and other reflective mechanisms): 

```python
class Foo:
    bar: str

    def __init__(self):
        self.baz: int = 0

if __name__ == '__main__':
    print(Foo.__annotations__)
```
will output `{'bar': <class 'str'>}`

Reply:
Great point! Updated 👍